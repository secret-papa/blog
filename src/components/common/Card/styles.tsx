import styled from "styled-components";

export const Root = styled.div`
  width: 320px;
  background-color: #ffffff;
  box-shadow: rgba(0, 0, 0, 0.04) 0px 4px 16px 0px;
  border-radius: 4px;
`;

export const Header = styled.div``;

export const CoverImageWrap = styled.div`
  height: 168px;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
  height: 100%;
  object-fit: cover;
`;

export const Body = styled.div`
  padding: 1rem;
`;

export const Title = styled.h4`
  font-size: 1rem;
  font-weight: bold;
  margin-bottom: 0.25rem;
  word-break: break-word;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  color: var(--color-grey-700);
`;

export const Description = styled.p`
  display: -webkit-box;
  height: 3.9375rem;
  font-size: 0.875rem;
  line-height: 1.5;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  color: var(--color-grey-600);
  margin-bottom: 1.5rem;
`;

export const Meta = styled.div`
  font-size: 0.75rem;
  color: var(--color-grey-500);
`;
