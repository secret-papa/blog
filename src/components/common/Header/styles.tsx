import styled from "styled-components";

export const Wrapper = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 64px;
`;

export const Header = styled.div`
  display: flex;
  width: 1024px;
`;
