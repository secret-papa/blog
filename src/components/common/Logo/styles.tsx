import styled from "styled-components";

// eslint-disable-next-line import/prefer-default-export
export const Text = styled.span`
  font-size: 1.4rem;
  color: #2e4f4f;
`;
