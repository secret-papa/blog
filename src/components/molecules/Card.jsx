import React from "react";
import PropTypes from "prop-types";
import { css } from "@emotion/react";

import { Text, Link, Image } from "../atoms";

const makeStyles = () => ({
  root: css`
    width: 26rem;
    height: 30rem;
    border-radius: 4px;
    overflow: hidden;
    background: #fff;
    box-shadow: 0px 4px 5px 0px rgba(0, 0, 0, 0.05);
  `,
  anchor: css`
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100%;
  `,
  cover: css`
    width: 100%;
    height: 14rem;
    background: #e6e6e6;
  `,
  description: ({ spacing }) => css`
    flex: 1;
    display: -webkit-box;
    display: -ms-flexbox;
    display: box;
    overflow: hidden;
    max-height: 5.7rem;
    margin-bottom: ${spacing(1.75)}rem;
    overflow: hidden;
    vertical-align: top;
    text-overflow: ellipsis;
    word-break: break-all;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
  `,
  body: ({ spacing }) => css`
    padding: ${spacing(2.5)}rem ${spacing(2)}rem;
    flex: 1;
    display: flex;
    flex-direction: column;
  `,
  title: ({ spacing }) => css`
    margin-bottom: ${spacing(1.75)}rem;
  `,
});

const Card = ({ cover, date, description, path, title, timeToRead }) => {
  const styles = makeStyles();

  return (
    <div css={styles.root}>
      <Link css={styles.anchor} to={path}>
        <figure css={styles.cover}>
          <Image src={cover} />
        </figure>
        <div css={styles.body}>
          <Text bold color="darkGray" component="p" customStyle={styles.title} variant="body1">
            {title}
          </Text>
          <div css={styles.description}>
            <Text variant="body2">{description}</Text>
          </div>
          <div>
            <Text color="lightGray2" variant="body2">
              {date} · ☕️ ️{timeToRead} min read
            </Text>
          </div>
        </div>
      </Link>
    </div>
  );
};

Card.defaultProps = {
  description: "",
};

Card.propTypes = {
  date: PropTypes.string.isRequired,
  description: PropTypes.string,
  cover: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  timeToRead: PropTypes.number.isRequired,
};

export default Card;
