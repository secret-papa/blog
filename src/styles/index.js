export { default as resetStyle } from "./reset";
export { default as theme } from "./theme";
