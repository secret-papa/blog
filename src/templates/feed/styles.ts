import styled from "styled-components";

export const Root = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: 100vh;
  background-color: #f8f9fa;
`;

export const Body = styled.main`
  width: 1024px;
`;

export const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  gap: 2rem;
`;
