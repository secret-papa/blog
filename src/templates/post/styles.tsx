import styled, { css } from "styled-components";
import { Link } from "gatsby";

export const Root = styled.div`
  margin-bottom: 3rem;
`;

export const Title = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: var(--color-grey-900);
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 5.5rem;
`;

export const Article = styled.article`
  width: 768px;
`;

export const CoverImageWrapper = styled.div`
  margin-top: 2rem;
`;

export const CoverImage = styled.img`
  display: block;
  width: 100%;
`;

export const MarkdownWrapper = styled.div`
  margin: 5rem 0 6rem;
`;

export const MarkdownBody = styled.div`
  & ol,
  ul {
    list-style: decimal;
  }
`;

export const NavigationButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  gap: 1rem;
`;

const leftDirectionCss = css`
  text-align: left;
`;
const rightDirectionCss = css`
  text-align: right;
`;

type Align = "left" | "right";

export const NavigationButton = styled(Link)<{ align?: Align }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 0.5rem;
  width: 100%;
  height: 4rem;
  background-color: #f8f9fa;
  padding-left: 1rem;
  padding-right: 1rem;
  box-shadow: rgba(0, 0, 0, 0.06) 0px 0px 4px 0px;
  border: none;
  cursor: pointer;
  ${props => (props.align === "right" ? rightDirectionCss : leftDirectionCss)}
`;

export const NavigationButtonCaption = styled.span`
  font-size: 0.75rem;
  color: var(--color-grey-600);
`;

export const NavigationButtonTitle = styled.h3`
  font-size: 1.125rem;
  font-weight: bold;
  color: var(--color-grey-700);
`;
