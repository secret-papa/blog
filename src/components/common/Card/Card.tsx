import React from "react";

import { Body, CoverImage, CoverImageWrap, Description, Header, Meta, Root, Title } from "./styles";

type CardProps = {
  title: string;
  description?: string;
  meta?: string[];
  coverImageUrl?: string;
  coverImageAlt?: string;
};

const Card = ({ title, description, meta, coverImageUrl, coverImageAlt }: CardProps) => (
  <Root>
    <Header>
      <CoverImageWrap>
        <CoverImage src={coverImageUrl} alt={coverImageAlt} />
      </CoverImageWrap>
    </Header>
    <Body>
      <Title>{title}</Title>
      <Description>{description}</Description>
      {meta && <Meta>{meta.join(" Â· ")}</Meta>}
    </Body>
  </Root>
);

export default Card;
