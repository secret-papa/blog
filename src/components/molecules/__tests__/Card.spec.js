import React from "react";

import { renderWithTheme } from "@tests";
import { theme } from "@src/styles";
import Card from "../Card";

describe("<Card />", () => {
  it("should show correctly", () => {
    const { container } = renderWithTheme(
      <Card
        cover="https://via.placeholder.com/150"
        date="xxxx-xx-xx"
        description="blah blah"
        path="/"
        title="Card title"
        timeToRead={3}
      />,
      theme,
    );
    expect(container).toMatchSnapshot();
  });
});
