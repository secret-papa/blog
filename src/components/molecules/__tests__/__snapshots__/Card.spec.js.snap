// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should show correctly 1`] = `
.emotion-0 {
  width: 26rem;
  height: 30rem;
  border-radius: 4px;
  overflow: hidden;
  background: #fff;
  box-shadow: 0px 4px 5px 0px rgba(0, 0, 0, 0.05);
}

.emotion-1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

.emotion-3 {
  width: 100%;
  height: 14rem;
  background: #e6e6e6;
}

.emotion-4 {
  display: block;
  width: 100%;
}

.emotion-5 {
  padding: 2rem 1.6rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-6 {
  color: #212121;
  font: bold 1.6rem/1 sans-serif;
  font-family: inherit;
  margin-bottom: 1.4rem;
}

.emotion-7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -ms-flexbox;
  display: box;
  overflow: hidden;
  max-height: 5.7rem;
  margin-bottom: 1.4rem;
  overflow: hidden;
  vertical-align: top;
  text-overflow: ellipsis;
  word-break: break-all;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-8 {
  color: #757575;
  font: normal 1.4rem/1.4 sans-serif;
  font-family: inherit;
}

.emotion-9 {
  color: #9E9E9E;
  font: normal 1.4rem/1.4 sans-serif;
  font-family: inherit;
}

<div>
  <div
    class="emotion-0"
  >
    <a
      class="emotion-1 emotion-2"
      href="/"
    >
      <figure
        class="emotion-3"
      >
        <img
          alt=""
          class="emotion-4"
          src="https://via.placeholder.com/150"
        />
      </figure>
      <div
        class="emotion-5"
      >
        <p
          class="emotion-6"
        >
          Card title
        </p>
        <div
          class="emotion-7"
        >
          <span
            class="emotion-8"
          >
            blah blah
          </span>
        </div>
        <div>
          <span
            class="emotion-9"
          >
            xxxx-xx-xx
             · ☕️ ️
            3
             min read
          </span>
        </div>
      </div>
    </a>
  </div>
</div>
`;
